backend/internal/mockprovider/handler_test.go

```
@@ -0,0 +1,264 @@
+	}
+	for index := range want {
+		if events[index] != want[index] {
+			t.Fatalf("events = %v", events)
+		}
+	}
+}
+
+func TestHandlerRejectsUnknownSurfaceAndModeMismatch(t *testing.T) {
+	handler := newTestHandler(t, "success.buffered", 1, nil)
+	unknown := httptest.NewRecorder()
+	handler.ServeHTTP(unknown, httptest.NewRequest(http.MethodPost, "/unknown", nil))
```

golangci-lint (2.12.2)

[error] net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext

(noctx)